internal/controllers/nmc_reconciler.go

```diff
@@ -424,14 +424,20 @@ func (h *nmcReconcilerHelperImpl) SyncStatus(ctx context.Context, nmcObj *kmmv1b
 		if status == nil {
 			status = &kmmv1beta1.NodeModuleStatus{
 				ModuleItem: kmmv1beta1.ModuleItem{
-					ImageRepoSecret:    nil,
-					Name:               modName,
-					Namespace:          modNamespace,
-					ServiceAccountName: "",
+					Name:      modName,
+					Namespace: modNamespace,
 				},
 			}
 		}
 
+		if irsName, err := getImageRepoSecretName(&p); err != nil {
+			return fmt.Errorf("error while looking for the imageRepoSecret volume: %v", err)
```

Comment:
we should record the error and continue instead of returning an error directly

Reply:
We control the Pod creation so this should never happen. However maybe we can make it a warning log line instead.

Reply:
warning log and continue works for me

Reply:
Done.